### ISSUE TITLE
Broken async middlewares

### ISSUE THREAD

**Issue by petersamokhin** [JakeFenley/koa-zod-router#18]
hello there, mate! thanks a lot for your effort making this library

there's an issue that is easy to reproduce:

the below code works (i.e. the response is `test`, in 5 sec):
```typescript
import Koa from 'koa'
import Router from '@koa/router'

const router = new Router()
const app = new Koa()

router.get('/foo', async ctx => {
  await new Promise(resolve => setTimeout(resolve, 5000))
  ctx.cookies.set('foo', 'bar')
  ctx.body = 'test'
})

app.use(router.routes())
app.listen(3000)
```

however, this doesn't work (the response "not found" is sent immediately, and in 5 seconds `ERR_HTTP_HEADERS_SENT` happens):
```typescript
import Koa from 'koa'
import { zodRouter } from 'koa-zod-router'

const router = zodRouter()
const app = new Koa()

router.get('/foo', async ctx => {
  await new Promise(resolve => setTimeout(resolve, 5000))
  ctx.cookies.set('foo', 'bar')
  ctx.body = 'test'
})

app.use(router.routes())
app.listen(3000)
```

**Comment by JakeFenley** [comment]
Hi Peter, thanks so much for catching this! I've just published 1.1.2 which i believe should fix the issue for you. 

hope you're enjoying the lib and if you have any additional feedback I would love to hear from you in the discussion section.

**Comment by petersamokhin** [comment]
@JakeFenley that was blazing fast, thank you for this! Everything is great so far, no more feedback 😄
